homeassistant/components/rainbird/switch.py

```diff
@@ -123,17 +103,20 @@ def __init__(
         coordinator: RainbirdUpdateCoordinator[States],
         rainbird: AsyncRainbirdController,
         zone: int,
-        time: int,
-        name: str,
+        duration_minutes: int,
+        serial_number: str,
+        device_info: DeviceInfo,
     ) -> None:
         """Initialize a Rain Bird Switch Device."""
         super().__init__(coordinator)
         self._rainbird = rainbird
         self._zone = zone
-        self._name = name
+        self._name = f"Sprinkler {zone}"
```

Comment:
Consider using the imported data to keep the user defined name as the default name for imported config entries during the deprecation period. That code can be removed when we remove the import as then the entities will have a unique id.

For entities where we don't use the imported name we can set `_attr_has_entity_name` to `True`.

Example:
https://github.com/home-assistant/core/blob/9e68117a338561b4e3b266684fa9ea3145b9f365/homeassistant/components/openexchangerates/sensor.py#L62-L69

Reply:
Do we need to story the current yaml in the config entry and keep it around and set this always? or is setting it the first time enough and it is preserved after that?

Reply:
The default name will change in the entity registry if we change the entity name here but the entity_id will be preserved. So there won't be a breaking change when we remove the migration code.

The config YAML is already stored in the config entry data in your current code as you pass the whole thing to the import step, right?

Reply:
OK. This now more completely handles importing names. I reduced the amount of data imported into the config entry to simplify the code in switch that will stick around, but it means it is a delta from before.

These are not yet using entity names, but I was thinking I may be able to actually pull in the actual device name or zone names from the devices in a follow up PR. Right now everything has really odd naming ('Raindelay', 'Rainsensor')